src/Akka.Hosting.Tests/Bugfix208Specs.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Actor.Dsl;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Xunit;
+
+namespace Akka.Hosting.Tests;
+
+public class Bugfix208Specs : TestKit.TestKit
+{
+    private class MyTestActor : ReceiveActor
+    {
+        public record SetData(string Data);
+
+        public record GetData();
+        
+        private string _data = string.Empty;
+        
+        public MyTestActor()
+        {
+            Receive<SetData>(s =>
+            {
+                _data = s.Data;
+            });
+            
+            Receive<GetData>(g =>
+            {
+                Sender.Tell(_data);
+            });
+        }
+    }
+    
+    private class TestActorKey{}
+    
+    private class MyBackgroundService : BackgroundService
+    {
+        private readonly IRequiredActor<TestActorKey> _testActor;
+
+        public MyBackgroundService(IRequiredActor<TestActorKey> requiredActor)
+        {
+           _testActor = requiredActor;
```

Comment:
So we tried to resolve the `IRequiredActor<TActor>.ActorRef` here, like users did in #208, but it won't work - if we block the constructor of a `BackgroundService` here it'll stop the `AkkaService` and another services from starting. It's too fragile of a fix for end-users, so instead we're going with throwing a clearer exception when an `ActorRegistry` miss occurs and offer an additional API on the `IRequiredActor<TActor>` interface to address it.